app/lib/actions/katello/repository/import_upload.rb

```diff
@@ -2,17 +2,36 @@ module Actions
   module Katello
     module Repository
       class ImportUpload < Actions::EntryAction
-        def plan(repository, upload_id, options = {})
+        middleware.use Actions::Middleware::KeepCurrentUser
+
+        def plan(repository, upload_ids, options = {})
           action_subject(repository)
-          unit_key = options.fetch(:unit_Key, {})
+          unit_keys = options.fetch(:unit_keys, {})
           generate_metadata = options.fetch(:generate_metadata, true)
-          import_upload = plan_action(Pulp::Repository::ImportUpload,
-                                      pulp_id: repository.pulp_id,
-                                      unit_type_id: repository.unit_type_id,
-                                      unit_key: unit_key,
-                                      upload_id: upload_id)
+          sync_capsule = options.fetch(:sync_capsule, true)
+          sequence do
+            concurrence do
+              upload_ids.zip(unit_keys) do |upload_id, unit_key|
+                import_upload = plan_action(Pulp::Repository::ImportUpload,
+                                            pulp_id: repository.pulp_id,
+                                            unit_type_id: repository.unit_type_id,
+                                            unit_key: unit_key,
+                                            upload_id: upload_id)
+
+                plan_action(FinishUpload, repository, :dependency => import_upload.output,
+                            generate_metadata: false)
+              end
+            end
+            plan_action(Katello::Repository::MetadataGenerate, repository) if generate_metadata
+            plan_self(repository_id: repository.id, sync_capsule: sync_capsule)
+          end
+        end
 
-          plan_action(FinishUpload, repository, :dependency => import_upload.output, :generate_metadata => generate_metadata)
+        def run
+          repository = ::Katello::Repository.find(input[:repository_id])
+          if input[:sync_capsule]
+            ForemanTasks.async_task(Katello::Repository::CapsuleGenerateAndSync, repository)
```

Comment:
We should likely do the same thing that repository sync is doing:  https://github.com/Katello/katello/blob/master/app/lib/actions/katello/repository/sync.rb#L50

otherwise you see this is the capsule is down:

```
$ hammer  repository  upload-content --id=39  --path ~/rpms/
Successfully uploaded file 'tfm-rubygem-jenkins_api_client-1.4.1-3.el7.noarch.rpm'.
Successfully uploaded file 'tfm-rubygem-mixlib-shellout-doc-2.2.1-3.el7.noarch.rpm'.
Failed to upload file 'tfm-rubygem-anemone-0.7.2-14.el7.noarch.rpm' to repository. Please check the file and try again.
```